discovery-infra/tests/test_discovery_ignition_override.py

```diff
@@ -1,37 +1,159 @@
+        except Exception as e:
+            logging.info("Got an exception while trying to update a cluster with unsupported discovery igniton")
+        else:
+            raise Exception("Expected patch_cluster_discovery_ignition to fail due to unsupported ignition version")
+
+    @pytest.mark.regression
+    def test_discovery_ignition_bad_format(self, api_client, node_controller, cluster):
+        """Create a discovery ignition override with old revision (2.2.0)
+        make sure patch API fail
+        """
+        # Define new cluster
+        cluster_id = cluster().id
+
+        ignition_override = copy.deepcopy(TestDiscoveryIgnition.base_ignition)
+        test_string = "not base 64b content"
+        override_path = "/ignition/test_discovery_ignition_bad_format"
+        ignition_override["storage"]["files"][0]["path"] = override_path
+        ignition_override["storage"]["files"][0]["contents"] = {"source": f"data:text/plain;base64,{test_string}"}
+
+        try:
+            api_client.patch_cluster_discovery_ignition(cluster_id, ignition_override)
+        except Exception as e:
+            logging.info("Got an exception while trying to update a cluster with unsupported discovery igniton")
+        else:
+            raise Exception("Expected patch_cluster_discovery_ignition to fail due to unsupported ignition version")
+
+    @pytest.mark.regression
+    def test_discovery_ignition_multiple_calls(self, api_client, node_controller, cluster):
+        """ Apply multiple discovery ignition overrides to the cluster.
+        Create a discovery ignition override and than create another one
+        Download the ISO
+        Start the nodes
+        Verify the override was applied
+        """
+        # Define new cluster
+        cluster_id = cluster().id
+
+        ignition_override = copy.deepcopy(TestDiscoveryIgnition.base_ignition)
+        override_path = "/ignition/test_discovery_ignition_multiple_calls_1"
+        ignition_override["storage"]["files"][0]["path"] = override_path
+        api_client.patch_cluster_discovery_ignition(cluster_id, ignition_override)
+
+        ignition_override_2 = copy.deepcopy(TestDiscoveryIgnition.base_ignition)
+        override_path_2 = "/ignition/test_discovery_ignition_multiple_calls_2"
+        ignition_override["storage"]["files"][0]["path"] = override_path_2
+        # Create another discovery ignition override
+        api_client.patch_cluster_discovery_ignition(cluster_id, ignition_override_2)
+
+        # Generate and download cluster ISO
+        self.generate_and_download_image(cluster_id=cluster_id, api_client=api_client)
+        # Boot nodes into ISO
+        hosts = node_controller.start_all_nodes()
+        # Wait until hosts are discovered and update host roles
+        self.wait_until_hosts_are_discovered(cluster_id=cluster_id, api_client=api_client)
+        # Verify override
+        self.validate_ignition_override(hosts, override_path)
```

Comment:
This will fail I think. We don't merge subsequent overrides into the original, it's a full replacement.